sdk/schemaregistry/schema-registry-avro/src/schemaRegistryAvroSerializer.ts

```diff
@@ -0,0 +1,226 @@
+   */
+  public autoRegisterSchemas: boolean;
+
+  // REVIEW: signature.
+  //
+  // - Better to serialize into a stream? I aborted that for now as I wanted to
+  //   do the simplest thing that could possibly work first to make sure there
+  //   were no blockers in our dependencies. I also wanted to get feedback on
+  //   what the API shape should be before diving into that.
+  //
+  // - This type should ultimately be able to implement a core ObjectSerializer
+  //   interface. Do we know what that would look like? Maybe it takes `any` as
+  //   the format-specific schema/type info? Or does it always take a
+  //   format-specific schema string?
+  //
+  //   The C#/Java approach of passing Type and assuming every serializer can
+  //   get its schema by reflecting on the type does not work for JavaScript. We
+  //   need to support arbitrary objects that match a schema.
+  //
+  //   Maybe each format expects a different property on this arg so that you
+  //   could at least pass enough info for multiple formats, and then your call
+  //   to ObjectSerializer is at least not tied to a single format?
+  //
+  // - Should we wrap all errors thrown by avsc to avoid having our exception //
+  //   contract being tied to its implementation details?
+  /**
+   * Serializes a value into a buffer.
+   *
+   * @param value The value to serialize.
+   * @param schema The Avro schema to use.
+   * @returns A new buffer with the serialized value
+   */
+  async serialize(value: any, schema: string): Promise<Buffer> {
+    const entry = await this.getSchemaByContent(schema);
+    const payload = entry.type.toBuffer(value);
+    const buffer = Buffer.alloc(PAYLOAD_OFFSET + payload.length);
+
+    buffer.writeUInt32BE(FORMAT_INDICATOR, 0);
+    buffer.write(entry.id, SCHEMA_ID_OFFSET, SCHEMA_ID_LENGTH, "utf-8");
+    payload.copy(buffer, PAYLOAD_OFFSET);
+    return buffer;
+  }
+
+  // REVIEW: signature. See serialize and s/serialize into/deserialize from/.
+  /**
+   * Deserializes a value from a buffer.
+   *
+   * @param buffer The buffer with the serialized value.
+   * @return The deserialized value.
+   */
+  async deserialize(buffer: Buffer): Promise<any> {
+    if (buffer.length < PAYLOAD_OFFSET) {
+      throw new Error("Buffer is too small to have the correct format.");
+    }
+
+    const format = buffer.readUInt32BE(0);
+    if (format !== FORMAT_INDICATOR) {
+      throw new Error(`Buffer has unknown format indicator: 0x${format.toString(16)}`);
+    }
+
+    const schemaIdBuffer = buffer.slice(SCHEMA_ID_OFFSET, PAYLOAD_OFFSET);
+    const schemaId = schemaIdBuffer.toString("utf-8");
+    const schema = await this.getSchemaById(schemaId);
+    const payloadBuffer = buffer.slice(PAYLOAD_OFFSET);
+
+    return schema.type.fromBuffer(payloadBuffer);
+  }
+
+  //
+  // Avoid hitting the schema registry on every call by caching schemas.
+  //
+  // Currently the cache can only be discarded by throwing away the serializer
```

Comment:
seems like having some kind of method to flush the cache would be reasonable

Reply:
I can certainly add a clearCache() now. As noted in the comment, we have https://github.com/Azure/azure-sdk-for-js/issues/10438 to figure out the long term caching story. Having it just leak until you throw away the serializer was what was deemed OK by the group for preview1. @ramya-rao-a What do you think should I add a clearCache() now to be replaced post-beta-1 with what we decide or go ahead as planned?

Reply:
I am fine waiting, sorry I didn't mean for my comment to sound blocking, I was just agreeing about what we could do eventually. :)